libs/ngx-slice-kit/src/lib/core/theme/theme.directive.spec.ts

```diff
@@ -33,11 +80,83 @@ describe('ThemeDirective', () => {
         //     themeService,
         //     layoutControl,
         // );
-        const directive = TestBed.inject(ThemeDirective);
+
         expect(directive).toBeTruthy();
     });
+
+    it('should create component', () => {
+        expect(component).toBeTruthy();
+    });
+
+    it('should #scoped be false by default', () => {
+        expect(directive.scoped).toBeFalse();
+    });
+
+    it('should #scoped set correct value from input', () => {
+        directive.scoped = true;
+        expect(directive.scoped).toBeTrue();
+    });
+
+    it('should #getElement return body element if #scoped is false', () => {
+        directive.scoped = false;
+        expect(directive.getElement().tagName).toEqual('BODY');
+    });
+
+    it('should #getElement return directive elementRef if #scoped is true', () => {
+        directive.scoped = true;
+        expect(directive.getElement()).toEqual({});
+    });
+
+    it('should #theme be undefined by default', () => {
+        expect(directive.theme).toBeUndefined();
+    });
+
+    it('should #theme has correct value from input if it was set', () => {
+        directive.theme = 'pinkblue';
+        expect(directive.theme).toEqual('pinkblue');
+    });
+
+    it('should be called #themeService.setTheme if #theme has value after init', () => {
+        const t = 'blue';
+        directive.theme = t;
+        spyOn(themeService, 'setTheme');
+
+        directive.ngOnInit();
+        expect(themeService.setTheme).toHaveBeenCalledWith(t);
+    });
+
+    // it('should', () => {});
+
+    // it('should', () => {});
+
+    // it('should', () => {});
+
+    // it('should', () => {});
+
+    // it('should', () => {});
+
+    // it('should', () => {});
+
+    // it('should', () => {});
+
+    // it('should', () => {});
+
+    // it('should', () => {});
+
```

Comment:
thi is what?

Reply:
templates for future descendants)) was deleted